### ISSUE TITLE
Add documentation

### ISSUE THREAD

**Issue by wowcut** [WowzaMediaSystems/wse-rest-library-php#55]
I can not find any documentation for this php classes anywhere.
The link to examples is not hepful, especially as [one](https://www.wowza.com/docs/how-to-use-php-to-make-requests-to-the-wowza-streaming-engine-rest-api) of them just links back to this repository.  
I can read the code - it´s more about the concepts. E.g. what is the thinking behind `WOWZA_SERVER_INSTANCE`, `WOWZA_VHOST_INSTANCE` and friends as constants? Assuming that there might be only one server seems very limiting - but maybe I do not get it right, and this is where documentation is missing.

**Comment by Maypeur** [comment]
Apparently, to retrieve the documentation you have to download it from your Wowza server : https://www.wowza.com/docs/how-to-access-documentation-for-wowza-streaming-engine-rest-api#more-resources 
I think they are crazy, 3 page of text to explain how to download a doc.  Why do they don't just provide a PDF ? Maybe it is too simple. 
--
For your question, WOWZA_SERVER_INSTANCE, WOWZA_VHOST_INSTANCE are to be used for the Settings class, if your read the source code, you will find that you can specify it with setServerInstance and setVhostInstance it must correspond with what you use in your Wowza server :
    $serverInstance = "_defaultServer_",
        $vhostInstance = "_defaultVHost_",
Are used by default and this is what is installed with a default Wowza Streaming Engine, so if you do not change your Wowza server configuration you can remove the constant in your PHP code.

**Comment by rogerlittin** [comment]
Those values should be 
```
$serverInstance = "_defaultServer_",
$vhostInstance = "_defaultVHost_",
```
They are normally the same across all servers, however, servers can have multiple vhosts so there may be cases where the VHost name may be different if your server has multiple VHosts.

@Maypeur, the articles you are referring to show how to enable and use the built-in swagger docs so you can test your api calls against the server.

**Comment by wowcut** [comment]
This issue is not resolved yet - you should close issues only after they have been resolved. 

If _you_ do not understand the issue or are not able to see the details, then it is proper behaviour to ask for clarification. Just simply closing an issue is a very disrespectful and destructive action.

The Swagger API of course I know and it might be a nice to have addition, but it is not - and was never meant, as a proper documentation. You can easily understand that by trying to study the details of the returned fields: many of these are not even described. These are just the very obvious missing parts. Very important additional information about the underlying concepts or abstract principles are not even possible to document with a Swagger API. Therefor this leaves too much room for interpretation.

If you still do not understand that please try to implent some kind of software that uses the Wowza REST API just with the Swagger API as docs - this practice will teach you a lot about how your customers feel when trying to do that. Now imagine you go and ask for details and somebody just closes the door before your nose - now you understand how your cutsomers feel when you just close this issue without showing the slightest effort of even trying to understand what it is about.

Please never close issues because _you believe_  that there is no issue or - even worse - because of company politics. That is a very toxic behaviour and a perfect way to destroy years of efforts of your marketing team. If you work for a company that would like to enter the "enterprise" space this must be reflected in a proper handling of all public activity, lead by respect and wisdom as first principles.

Again: auto-generated API-doc does **NOT** replace a hand-written documentation.

**Comment by wowcut** [comment]
One more reason why this issue is not resolved: the first link below "Usage" named "[How to use PHP to make requests to the Wowza Streaming Engine REST API](https://www.wowza.com/docs/how-to-use-php-to-make-requests-to-the-wowza-streaming-engine-rest-api)" still points to a website that does only contain a link back to this repository - this is not helpful at all. Did you even try to reproduce the reported issue?

**Comment by Wowzadocs** [comment]
Hello and thanks for your comments. I'm the Wowza doc manager and I'm sorry for the poor experience you've had with our documentation. You're correct: our Wowza Streaming Engine REST API reference documentation is difficult to access and incomplete. We are aware of these issues and they're in our backlog as work we hope to address in 2019. We'll remove the circular links back to our website as soon as possible. If it's helpful, we do have cURL query examples at https://www.wowza.com/docs/wowza-streaming-engine-rest-api-query-examples-curl. Thank you for your patience.
